src/main/java/com/scaleunlimited/flinkcrawler/functions/ParseFunction.java

```diff
@@ -78,31 +94,20 @@ public void flatMap(FetchedUrl fetchedUrl,
 
             @Override
             public int compare(ExtractedUrl o1, ExtractedUrl o2) {
-                return (int) (o1.getScore() - o2.getScore());
+                return (int) (o2.getScore() - o1.getScore());
             }
         });
         
         int count = 0;
         for (ExtractedUrl outlink : extractedUrls) {
             String message = String.format("Extracted '%s' from '%s'", outlink.getUrl(),
```

Comment:
Minor improvement - since we extract so many outlinks, it would be best to wrap this in a `if (LOGGER.isTraceEnabled()) { ... }` block.